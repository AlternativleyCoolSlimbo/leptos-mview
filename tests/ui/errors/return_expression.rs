use leptos_mview::view;

fn main() {
    // should not get an "unexpected end of macro invocation"
    let expr = view! {
        div class=;
    };
}