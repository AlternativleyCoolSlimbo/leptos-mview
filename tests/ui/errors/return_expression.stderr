error: expected value after =
 --> tests/ui/errors/return_expression.rs:6:19
  |
6 |         div class=;
  |                   ^

error[E0277]: the trait bound `MissingValueAfterEq: IntoAttribute` is not satisfied
 --> tests/ui/errors/return_expression.rs:6:19
  |
5 |       let expr = mview! {
  |  ________________-
6 | |         div class=;
  | |                   ^ the trait `std::ops::Fn<()>` is not implemented for `MissingValueAfterEq`, which is required by `MissingValueAfterEq: IntoAttribute`
7 | |     };
  | |_____- required by a bound introduced by this call
  |
  = help: the following other types implement trait `IntoAttribute`:
            &'static str
            &std::string::String
            Arguments<'_>
            Cow<'static, str>
            Nonce
            Oco<'static, str>
            Rc<str>
            TextProp
          and $N others
  = note: required for `MissingValueAfterEq` to implement `IntoAttribute`
note: required by a bound in `leptos::HtmlElement::<El>::attr`
 --> $CARGO/leptos_dom-0.6.12/src/html.rs
  |
  |     pub fn attr(
  |            ---- required by a bound in this associated function
...
  |         attr: impl IntoAttribute,
  |                    ^^^^^^^^^^^^^ required by this bound in `HtmlElement::<El>::attr`
